### ISSUE TITLE
Add DSA to CoreFX

### ISSUE THREAD

**Issue by bartonjs** [dotnet/corefx#4813]
When added there should be adequate support for implementation hiding, and it should support FIPS 186-3's addition of SHA-2 algorithms.

**Comment by RussellTaylor83** [comment]
Hello, can you tell me when we might see DSACryptoServiceProvider please? If there are no plans for it or it's a way off I may try and find a way to do it without .Net, which would be a shame as I'm just trying to port my existing work to .Net core. I can probably achieve the same thing using just BouncyCastle, but at present I cannot get that to run against dotnet 5.4 either.

Thank you :+1:

**Comment by bartonjs** [comment]
@RussellTaylor83 Sorry for the long window of non-response; I must have missed this notification when it came in.  The DSA base class got some improvement in .NET 4.6.2 (currently in preview) and I wanted those to be present before we started moving things to .NET Core.

As of now it is slated for our 1.1 release, tentatively targeted as November.  I expect it to be done before the heat is off of the North American summer; but that's definitely not a commitment.

**Comment by bartonjs** [comment]
@RussellTaylor83 Bah, hit comment too soon.

When porting to .NET Core (and in your existing code) you should avoid talking about DSACryptoServiceProvider, if you can.  That type is for a Windows CAPI implementation of DSA/DSS, You should just use the DSA base class (which should be doable in net462, provided you're not using named/persisted keys).

**Comment by danmosemsft** [comment]
Both of
```
T:System.Security.Cryptography.DSA
T:System.Security.Cryptography.DSAParameters
```

**Comment by steveharter** [comment]
To be clear, this work involves four areas:
1) S.S.C.Algorithms to contain portable API (private implementation taken from Cng and OpenSsl)
2) S.S.C.Cng for windows\Cng specific functionality
3) S.S.C.OpenSsl for linux\osx OpenSsl specific functionality
4) S.S.C.Csp to contain the Csp version (calls into Algorithms Cng)

**Comment by RussellTaylor83** [comment]
Thank you all.

**Comment by steveharter** [comment]
The four areas are implemented

**Comment by bartonjs** [comment]
@steveharter I don't think we did the DSACertificateExtensions class (GetDSAPublicKey, GetDSAPrivateKey)... did we?

**Comment by steveharter** [comment]
@bartonjs no I didn't see that. Added #11802
